rflx/pyrflx/typevalue.py

```diff
@@ -886,10 +878,15 @@ def set_refinement(fld: MessageValue.Field, fld_name: str) -> None:
                         fld.typeval.set_refinement(ref.sdu)
 
         def check_outgoing_condition_satisfied() -> None:
-            if all(
-                self.__simplified(o.condition) == FALSE
-                for o in self._type.outgoing(Field(field_name))
-            ):
+            simplified = [
+                self.__simplified(o.condition) for o in self._type.outgoing(Field(field_name))
+            ]
+            unresolved = [o for o in simplified if o not in (FALSE, TRUE)]
+            assert not unresolved, (  # pragma: no cover
+                f"unresolved field conditions in {self.model.name}.{field_name}:"
+                f" {', '.join([str(o) for o in unresolved])}"
+            )
+            if all(o == FALSE for o in simplified):
```

Comment:
Why is `pragma: no cover` required here? Even if the assertion is never false in any test, the statement should be covered.

Reply:
When I remove it, I get:

```
Name                       Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------------------
rflx/pyrflx/typevalue.py     826      0    352      1    99%   885->exit
----------------------------------------------------------------------
TOTAL                       8825      0   3838      1    99%
```

This is consistent with my understanding of branch coverage. I wonder why we didn't have this problem elsewhere. I don't see a good way to trigger this assertion, though, as it is only a precaution and the code is very convoluted which makes it hard to find the right place to monkeypatch a failure in.
If you have a good idea on how to deal with it without that pragma, I'd be happy to hear it.

Reply:
This seems to be a bug in the coverage calculation. Usually, an assertion does not lead to a new branch. If the assertion is changed into a single line statement, the missing branch vanishes. The issue is similar to nedbat/coveragepy#515, although the cause seems to be different.

Reply:
We even [had the same issue in the past](https://github.com/Componolit/RecordFlux/pull/381#discussion_r463857067).

The list comprehension seems to be the culprit (although we have something similar in the assertion below). I reworked the code to construct the error message string before the assertion. This fixes the coverage issue (but has the downside that we construct the string even with assertions off).

Also rebased to `main`.